client/state/data-layer/wpcom/user-settings/test/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { useSandbox } from 'test/helpers/use-sinon';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	updateUserSettings,
+	clearUnsavedUserSettings,
+} from 'state/user-settings/actions';
+import {
+	requestUserSettings,
+	receiveUserSettings,
+	saveUserSettings,
+	receiveSavedUserSettings,
+} from '../';
+
+describe( 'wpcom-api', () => {
+	let dispatch, next;
+
+	useSandbox( sandbox => {
+		dispatch = sandbox.spy();
+		next = sandbox.spy();
+	} );
+
+	describe( 'user settings request', () => {
+		describe( '#requestUserSettings', () => {
+			it( 'should dispatch HTTP GET request to me/settings endpoint', () => {
+				const action = { type: 'DUMMY' };
+
+				requestUserSettings( { dispatch }, action, next );
+
+				expect( dispatch ).to.have.been.calledOnce;
+				expect( dispatch ).to.have.been.calledWith( http( {
+					apiVersion: '1.1',
+					method: 'GET',
+					path: '/me/settings',
+				}, action ) );
+			} );
+
+			it( 'should pass the original action along the middleware chain', () => {
+				const action = { type: 'DUMMY' };
+
+				requestUserSettings( { dispatch }, action, next );
+
+				expect( next ).to.have.been.calledWith( action );
+			} );
+		} );
+
+		describe( '#receiveUserSettings', () => {
+			it( 'should dispatch user settings update', () => {
+				const action = { type: 'DUMMY' };
+
+				receiveUserSettings( { dispatch }, action, next, {
+					description: 'foo &amp; bar',
+					language: 'qix',
+				} );
+
+				expect( dispatch ).to.have.been.calledOnce;
+				expect( dispatch ).to.have.been.calledWith( updateUserSettings( {
+					description: 'foo & bar',
+					language: 'qix',
+				} ) );
+			} );
```

Comment:
nice tests here! I appreciate how you are using _unit_ tests and letting the other layers (selectors, dispatcher) to test their own behaviors and leaving that coupling out of this file